rclrs/src/lib.rs

```diff
@@ -147,3 +152,136 @@ pub fn create_node(context: &Context, node_name: &str) -> Result<Node, RclrsErro
 pub fn create_node_builder(context: &Context, node_name: &str) -> NodeBuilder {
     Node::builder(context, node_name)
 }
+
+/// Extract non-ROS arguments from program's input arguments.
+///
+/// `args` is expected to be the input arguments of the program (passed via [`env::args()`]).
+/// According to rcl documentation, ROS arguments are between `--ros-args` and `--` arguments.
+/// Everything else is considered as non-ROS and left unparsed. Extracted non-ROS args are
+/// returned in the order that they appear (see example below).
+///
+/// # Example
+/// ```
+/// let input_args: Vec<String> = vec![
+///             "arg1", "--ros-args", "some", "args", "--", "arg2"
+///         ]
+///         .into_iter()
+///         .map(|x| x.to_string())
+///         .collect();
+/// let non_ros_args = rclrs::extract_non_ros_args(input_args).unwrap();
+/// assert_eq!(non_ros_args.len(), 2);
+/// assert_eq!(non_ros_args[0], "arg1");
+/// assert_eq!(non_ros_args[1], "arg2");
+/// ```
+pub fn extract_non_ros_args(
+    args: impl IntoIterator<Item = String>,
+) -> Result<Vec<String>, RclrsError> {
+    // SAFETY: No preconditions for this function
```

Comment:
Nitpick: We've used this phrasing elsewhere, would be ideal to use it here too for consistency:

```suggestion
    // SAFETY: Getting a zero-initialized value is always safe.
```